### DISCUSSION TITLE
a beginner looking for some help

### DISCUSSION THREAD

**Discussion by elhejj888** [mastodon/mastodon#24513] [General]
Hi there,

I am a software engineering student and I am interested in learning how to create a basic decentralized social media website. I am a beginner in web3 and decentralization programming methods, so I would appreciate any details or guidance you can provide to get me started.

I have a basic understanding of programming languages such as Python, JavaScript, and Solidity. However, I am not sure where to begin with creating a decentralized social media website. Can you recommend any resources or tutorials that can help me get started with this project?

I would also appreciate any advice on the best programming languages, frameworks, and tools to use for this project. Any guidance or tips would be greatly appreciated.

Thank you in advance for your help!

**Comment by fishcharlie** [comment]
Why not start with looking through the issues on this repo and solving some of those? Learning about ActivityPub. Reviewing pull requests. Truly understand the basics and protocols, before jumping into building something on your own.

Tutorials don't exist to create something as complicated as a "decentralized social media website". There are over 13k commits on this repo (not to mention all the commits on the other related repos). It's not something a single tutorial can encompass.

Finally, you are probably asking on the wrong place. This is a Discussion page for Mastodon. People here came for Mastodon Discussions, not to help you start your own alternative.

**Reply by elhejj888** [reply]
Yes, thank you for your help. I didn't mean anything negative, I just tried to find some help in different places like Twitter, Facebook, and Reddit, but no one answered. I thought maybe it would be a good idea to ask here since the entire Mastodon community has a shared passion.

My question wasn't about creating an alternative to Mastodon; in fact, I really like the way it works and the fact that it's all open-source, making it accessible to different researchers. All I wanted was to create a decentralized application for my school project, and the best idea I had was a social media platform.

Anyway, thank you for your help, and I will try to apply your valuable advice.
